### ISSUE TITLE
Applying admin-user addon with --openshift-version 3.10.0-rc.0 results in 'No such file or directory'

### ISSUE THREAD

**Issue by donovanmuller** [minishift/minishift#2604]
### General information

  * Minishift version: 

```
$ minishift version
minishift v1.21.0+a8c8b37
```

  * OS: Linux / macOS / Windows

macOS

  * Hypervisor: KVM / xhyve / Hyper-V / VirtualBox

xhyve

### Steps to reproduce

  1. Start a `minishift` instance with `--openshift-version 3.10.0-rc.0`
  2. Wait for successful start
  3. Try to apply the `admin-user` addon with `minishift addons apply admin-user`

### Expected

`admin-user` addon to be applied successfully

### Actual

```
$ minishift addons apply admin-user
Cannot get the OpenShift master configuration: ssh command error:
command : docker exec -t 282cd17f45d5 cat /var/lib/origin/openshift.local.config/master/master-config.yaml
err     : exit status 1
output  : cat: /var/lib/origin/openshift.local.config/master/master-config.yaml: No such file or directory
```

**Comment by praveenkumar** [comment]
@donovanmuller I tried as following and it works as expected.

```
$ ./minishift start --openshift-version 3.10.0-rc.0
-- Starting profile 'minishift'
-- Check if deprecated options are used ... OK
-- Checking if https://github.com is reachable ... OK
[...]
Login to server ...
Creating initial project "myproject" ...
Server Information ...
OpenShift server started.

The server is accessible via web console at:
    https://192.168.64.3:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin

$  $ ./minishift status
Minishift:  Running
Profile:    minishift
OpenShift:  Running (openshift v3.10.0-rc.0+745de5b-46)
DiskUsage:  21% of 19G (Mounted On: /mnt/vda1)
CacheUsage: 2.109 GB (used by oc binary, ISO or cached images)

$ ./minishift addon apply admin-user
-- Applying addon 'admin-user':..

 $ oc whoami
developer

10:43 $ oc login -u admin -p admin
Login successful.

You have access to the following projects and can switch between them with 'oc project <projectname>':

    default
    kube-dns
    kube-proxy
    kube-public
    kube-system
  * myproject
    openshift
    openshift-apiserver
    openshift-controller-manager
    openshift-core-operators
    openshift-infra
    openshift-node
    openshift-web-console

Using project "myproject".

10:43 $ oc whoami
admin
```

**Comment by LalatenduMohanty** [comment]
I am reproduce it on macOS+ Xhyve on latest master .

```
$ minishift start --openshift-version v3.10.0-rc.0
-- Starting profile 'test'
-- Check if deprecated options are used ... OK
-- Checking if https://github.com is reachable ... OK
-- Checking if requested OpenShift version 'v3.10.0-rc.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.10.0-rc.0' is supported ... OK
-- Checking if requested hypervisor 'xhyve' is supported on this platform ... OK
-- Checking if xhyve driver is installed ...
   Driver is available at /usr/local/bin/docker-machine-driver-xhyve
   Checking for setuid bit ... OK
-- Checking the ISO URL ... OK
-- Downloading OpenShift binary 'oc' version 'v3.10.0-rc.0'
 23.83 MiB / 23.83 MiB [==============================================================================================================================================] 100.00% 0s-- Downloading OpenShift v3.10.0-rc.0 checksums ... OK
-- Checking if provided oc flags are supported ... OK
-- Starting the OpenShift cluster using 'xhyve' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    2 GB
   vCPUs :    2
   Disk size: 20 GB
-- Starting Minishift VM ............... OK
-- Checking for IP address ... OK
-- Checking for nameservers ... OK
-- Checking if external host is reachable from the Minishift VM ...
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ...
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 1% used OK
-- OpenShift cluster will be configured with ...
   Version: v3.10.0-rc.0
-- Pulling the Openshift Container Image ............... OK
-- Copying oc binary from the OpenShift container image to VM ... OK
-- Starting OpenShift cluster ..............................................................
Getting a Docker client ...
Checking if image openshift/origin-control-plane:v3.10 is available ...
Pulling image openshift/origin-control-plane:v3.10
Pulled 2/5 layers, 44% complete
Pulled 3/5 layers, 72% complete
Pulled 4/5 layers, 83% complete
Pulled 5/5 layers, 100% complete
Extracting
Image pull complete
Pulling image openshift/origin-cli:v3.10
Pulled 1/4 layers, 27% complete
Pulled 2/4 layers, 70% complete
Pulled 3/4 layers, 81% complete
Pulled 4/4 layers, 100% complete
Extracting
Image pull complete
Pulling image openshift/origin-node:v3.10
Pulled 5/6 layers, 84% complete
Pulled 6/6 layers, 100% complete
Extracting
Image pull complete
Checking type of volume mount ...
Determining server IP ...
Using public hostname IP 192.168.64.2 as the host IP
Checking if OpenShift is already running ...
Checking for supported Docker version (=>1.22) ...
Checking if insecured registry is configured properly in Docker ...
Checking if required ports are available ...
Checking if OpenShift client is configured properly ...
Checking if image openshift/origin-control-plane:v3.10 is available ...
Starting OpenShift using openshift/origin-control-plane:v3.10 ...
I0724 01:35:11.092245    2014 config.go:42] Running "create-master-config"
I0724 01:35:13.415835    2014 config.go:46] Running "create-node-config"
I0724 01:35:14.415662    2014 flags.go:30] Running "create-kubelet-flags"
I0724 01:35:15.038383    2014 run_kubelet.go:48] Running "start-kubelet"
I0724 01:35:15.181151    2014 run_self_hosted.go:172] Waiting for the kube-apiserver to be ready ...
I0724 01:36:17.188355    2014 interface.go:26] Installing "kube-proxy" ...
I0724 01:36:17.189156    2014 interface.go:26] Installing "kube-dns" ...
I0724 01:36:17.189167    2014 interface.go:26] Installing "openshift-apiserver" ...
I0724 01:36:17.189194    2014 apply_template.go:83] Installing "openshift-apiserver"
I0724 01:36:17.190608    2014 apply_template.go:83] Installing "kube-proxy"
I0724 01:36:17.191576    2014 apply_template.go:83] Installing "kube-dns"
I0724 01:36:19.326483    2014 interface.go:41] Finished installing "kube-proxy" "kube-dns" "openshift-apiserver"
I0724 01:37:10.358610    2014 run_self_hosted.go:224] openshift-apiserver available
I0724 01:37:10.359302    2014 interface.go:26] Installing "openshift-controller-manager" ...
I0724 01:37:10.359355    2014 apply_template.go:83] Installing "openshift-controller-manager"
I0724 01:37:13.152096    2014 interface.go:41] Finished installing "openshift-controller-manager"
Adding default OAuthClient redirect URIs ...
Adding centos-imagestreams ...
Adding router ...
Adding sample-templates ...
Adding persistent-volumes ...
Adding web-console ...
Adding registry ...
I0724 01:37:13.228902    2014 interface.go:26] Installing "centos-imagestreams" ...
I0724 01:37:13.228945    2014 interface.go:26] Installing "openshift-router" ...
I0724 01:37:13.228982    2014 interface.go:26] Installing "sample-templates" ...
I0724 01:37:13.229019    2014 interface.go:26] Installing "persistent-volumes" ...
I0724 01:37:13.229025    2014 interface.go:26] Installing "openshift-web-console-operator" ...
I0724 01:37:13.229033    2014 interface.go:26] Installing "openshift-image-registry" ...
I0724 01:37:13.231902    2014 apply_list.go:68] Installing "centos-imagestreams"
I0724 01:37:13.233481    2014 apply_template.go:83] Installing "openshift-web-console-operator"
I0724 01:37:13.234564    2014 interface.go:26] Installing "sample-templates/jenkins pipeline ephemeral" ...
I0724 01:37:13.234632    2014 interface.go:26] Installing "sample-templates/sample pipeline" ...
I0724 01:37:13.234650    2014 interface.go:26] Installing "sample-templates/mongodb" ...
I0724 01:37:13.234668    2014 interface.go:26] Installing "sample-templates/mariadb" ...
I0724 01:37:13.234686    2014 interface.go:26] Installing "sample-templates/mysql" ...
I0724 01:37:13.234700    2014 interface.go:26] Installing "sample-templates/postgresql" ...
I0724 01:37:13.234714    2014 interface.go:26] Installing "sample-templates/dancer quickstart" ...
I0724 01:37:13.234829    2014 interface.go:26] Installing "sample-templates/django quickstart" ...
I0724 01:37:13.234867    2014 interface.go:26] Installing "sample-templates/cakephp quickstart" ...
I0724 01:37:13.234878    2014 interface.go:26] Installing "sample-templates/nodejs quickstart" ...
I0724 01:37:13.234886    2014 interface.go:26] Installing "sample-templates/rails quickstart" ...
I0724 01:37:13.234983    2014 apply_list.go:68] Installing "sample-templates/rails quickstart"
I0724 01:37:13.235276    2014 apply_list.go:68] Installing "sample-templates/jenkins pipeline ephemeral"
I0724 01:37:13.235390    2014 apply_list.go:68] Installing "sample-templates/sample pipeline"
I0724 01:37:13.235492    2014 apply_list.go:68] Installing "sample-templates/mongodb"
I0724 01:37:13.235635    2014 apply_list.go:68] Installing "sample-templates/mariadb"
I0724 01:37:13.235734    2014 apply_list.go:68] Installing "sample-templates/mysql"
I0724 01:37:13.235826    2014 apply_list.go:68] Installing "sample-templates/postgresql"
I0724 01:37:13.235928    2014 apply_list.go:68] Installing "sample-templates/dancer quickstart"
I0724 01:37:13.236025    2014 apply_list.go:68] Installing "sample-templates/django quickstart"
I0724 01:37:13.236116    2014 apply_list.go:68] Installing "sample-templates/cakephp quickstart"
I0724 01:37:13.236262    2014 apply_list.go:68] Installing "sample-templates/nodejs quickstart"
I0724 01:37:27.759618    2014 interface.go:41] Finished installing "sample-templates/jenkins pipeline ephemeral" "sample-templates/sample pipeline" "sample-templates/mongodb" "sample-templates/mariadb" "sample-templates/mysql" "sample-templates/postgresql" "sample-templates/dancer quickstart" "sample-templates/django quickstart" "sample-templates/cakephp quickstart" "sample-templates/nodejs quickstart" "sample-templates/rails quickstart"
Login to server ...
I0724 01:38:13.613919    2014 interface.go:41] Finished installing "centos-imagestreams" "openshift-router" "sample-templates" "persistent-volumes" "openshift-web-console-operator" "openshift-image-registry"
Creating initial project "myproject" ...
Server Information ...
OpenShift server started.

The server is accessible via web console at:
    https://192.168.64.2:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin

```

```
$ minishift addons apply admin-user
Cannot get the OpenShift master configuration: ssh command error:
command : docker exec -t d707fd1773a2 cat /var/lib/origin/openshift.local.config/master/master-config.yaml
err     : exit status 1
output  : cat: /var/lib/origin/openshift.local.config/master/master-config.yaml: No such file or directory

```
The file is not present in the location 

```
$ minishift ssh "ls /var/lib/origin/openshift.local.config/master/"
ls: cannot access /var/lib/origin/openshift.local.config/master/: No such file or directory
Command failed: exit status 2
```

**Comment by LalatenduMohanty** [comment]
For me I can only reproduce this issue when I am running on non default profile. On default profile this is not reproducible.

**Comment by LalatenduMohanty** [comment]
I can manually run the commands without any issue. You can use this as workaround also. 

```
$ oc login -u system:admin
$ oc create user admin --full-name=admin
$ oc adm policy add-cluster-role-to-user cluster-admin admin

```

**Comment by LalatenduMohanty** [comment]
Resolved via #2627
